### ISSUE TITLE
No route to host while connecting with adb connect

### ISSUE THREAD

**Issue by adarshbalu** [Genymobile/scrcpy#1341]
I checked a lot on stack overflow but still, I can't get this working.
This shows every time I use `adb connect `
`unable to connect to 192.168.1.100:5555: No route to host`

**Comment by rom1v** [comment]
Are your device and your computer on the same network?

**Comment by adarshbalu** [comment]
> Are your device and your computer on the same network?

Yes . Connected in the same wifi network. I tried restarting both my phone and my pc.
I also tried adb kill-server.
I looked into a lot of answers on the Internet still no solution.

**Comment by rom1v** [comment]
Did you `adb tcpip 5555`?

**Comment by adarshbalu** [comment]
> Did you `adb tcpip 5555`?

Yes i did. I'll show you the complete screenshot.

**Comment by rom1v** [comment]
> I'll show you the complete screenshot.

(or better, copy-paste the text from your terminal :wink:)

**Comment by adarshbalu** [comment]
> > I'll show you the complete screenshot.
> 
> (or better, copy-paste the text from your terminal )

`adb tcpip 5555`
* daemon not running; starting now at tcp:5037
* daemon started successfully

`adb connect 192.168.1.200`
unable to connect to 192.168.1.200:5555: No route to host

**Comment by adarshbalu** [comment]
> > I'll show you the complete screenshot.
> 
> (or better, copy-paste the text from your terminal )

Hey

**Comment by rom1v** [comment]
I don't know. If your device is connected over wifi, if you use its IP address and if your computer is on the same network, it should work.

https://stackoverflow.com/questions/33462720/adb-unable-to-connect-to-192-168-1-105555

**Comment by adarshbalu** [comment]
> I don't know. If your device is connected over wifi, if you use its IP address and if your computer is on the same network, it should work.
> 
> https://stackoverflow.com/questions/33462720/adb-unable-to-connect-to-192-168-1-105555

I tried the same post and still does not work. Even kde connect is not working. The device is not showing.

**Comment by dimashn21** [comment]
i have same problem

**Comment by dimashn21** [comment]
i know,, you must check ip address your phone again..

**Comment by williamjnrkdd** [comment]
So my situation was that i restarted it in tcp mode but still couldn't connect, i had the same error ie no route to host, so i tried pinging it but i couldn't find it either even though it was on the arp table. so what i noticed was that when i connect it by usb, it pings successfully. so i  had to do the ``` adb connect ip``` before i could unplug the usb.

**Comment by fusionneur** [comment]
I created this tool to ease wireless connectivity. Check it out and let me know of your experience.

https://github.com/fusionneur/ethernally

**Comment by JuJu2181** [comment]
Any one got solutions to this problem?

**Comment by zrhhub** [comment]
**i'm also facing the same problem.!!**

**Comment by Abhishek-EE** [comment]
I was having the same issue but after looking at @williamjnrkdd I decided to look at arp table and the HWaddress was incomplete. After digging in a bit it turns out the MAC address on my android was randomized which made it impossible to establish connection with even when I was using ping. Ideally go to settings search for Wifi Mac Address and disable the random mac address and use the phone address instead. Then you also need to go to developer option and follow this post https://developer.android.com/studio/run/device for adding your linux system to your android debugging tools. and then try pinging the ip of your phone from your machine and it should work. @coddrikaz hope this helps

**Comment by richardwojc** [comment]
how about with usb

**Comment by uvlad7** [comment]
It worked for some reason when I reserved an IP for watches' MAC in my router DHCP settings.

**Comment by VishnuRathore98** [comment]
Was facing the same issue, the following worked for me:
1. Connect your device with usb(make sure usb debugging is enabled).
2. Execute: `adb usb`.
3. Execute: `adb tcpip 5555` (this will restart in TCP mode on port 5555).
4. Execute: `adb connect 192.168.1.3:5555` (this will connect your device using IP:PORT).

Now, from next time you just need to execute the last command using proper device's IP and 5555 port.

**Comment by jamescl131** [comment]
Same issue, but running the following resolved it for me. 

adb kill-server
adb start-server
adb usb 
adb tcpip 5555 
scrcpy --tcpip=<IP Address> -Sw

(I wanted Sw switches, these are optional for keep screen unlocked and turn screen off when you're connected)

**Comment by cHAuHaNz** [comment]
This worked for me as well:

> adb kill-server
> adb start-server
> adb usb
> adb tcpip 5555
> adb connect <DEVICE_LOCAL_IP>

**Comment by zailaib** [comment]
0. same wifi connectd of you phone and pc
1. set usb debug mode on your android device first
2. connect the Cable between phone and you pc
3. run commands : adb kill-server && adb start-server && adb tcpip 5555 && adb connect <you phone ip addr>
4. now you can remove connected of the USB cable

**do not run `adb usb`**

**Comment by y1n0** [comment]
I've notice this happens when there is already an android emulator running on the machine? maybe starting an emulator somehow cause adb to use an internel network between the machine and the emulator? any lead on how to start debugging this?

**Comment by JiaMingW** [comment]
Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.

This is work for me!

**Comment by tidianeb5** [comment]
> Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.
> 
> This is work for me!

this work for me too. thanks

**Comment by lakshaycodeapex** [comment]
command: "adb kill-server && adb tcpip 5037 && adb connect 192.168.1.2:5037"

Output:
* daemon not running; starting now at tcp:5037
* daemon started successfully
restarting in TCP mode port: 5037
connected to 192.168.1.2:5037

**Comment by rom1v** [comment]
(even if it works, it's a bit weird to use 5037, which is the local port for the adb daemon, the device port is typically 5555)

**Comment by yogithesymbian** [comment]
**I’ve been a developer for over six years and have solved similar issues using the solutions mentioned above.** 

**However, I faced the same problem again today and spent three hours troubleshooting.** 

Then I remembered that I had set Cloudflare/DNS to 1.1.1.1 to bypass a blocked website two days ago. After removing those DNS settings (on macOS) by going to Wi-Fi -> DNS -> and deleting the custom settings, I tried the solutions again—and it worked like a charm! lol.

**Comment by iammofidul** [comment]
> Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.
> 
> This is work for me!

**Comment by rin1652** [comment]
> Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.
> 
> This is work for me!

this is work for me! I toggle android studio app

**Comment by AlexArms** [comment]
As another alternative solution to this problem for anyone looking, when I had Android Studio open on my machine I'd get this error. I'd need to close studio and then use adb to connect to devices before starting it. Annoying but simple at least.

**Comment by mountainn-dev** [comment]
> Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.
> 
> This is work for me!

what a genious! thank you 👍

**Comment by zaidmian** [comment]
> As another alternative solution to this problem for anyone looking, when I had Android Studio open on my machine I'd get this error. I'd need to close studio and then use adb to connect to devices before starting it. Annoying but simple at least.

This is the only thing that worked for me. Thank you!!

**Comment by GaganRaghunath** [comment]
On a Macbook go to **System Settings > Privacy & Security > Local Network** and enable Android Studio to access local network.
Then try the **adb** commands.
Worked for me!!!

**Comment by MdAfzaalkhan** [comment]
> Go to System Settings > Privacy Security > Local Network, ensure the toggle is on for your terminal app.
> 
> This is work for me but toggle on the Android studio my terminal was already ON so i tried enabling android studio and it worked

**Comment by mrArtCore** [comment]
> On a Macbook go to **System Settings > Privacy & Security > Local Network** and enable Android Studio to access local network. Then try the **adb** commands. Worked for me!!!

Saviour! ✌️
How could I forget to check that in the first place 🥲

**Comment by naratteu** [comment]
저는 며칠동안 틀어둔 맥북을 재부팅 하여 해결했습니다.
